extcap_ot.py

```diff
@@ -0,0 +1,205 @@
+    args.append((Config.TAP.value, '--tap', 'IEEE 802.15.4 TAP (only for Wireshark3.0 and later)', 'IEEE 802.15.4 TAP', 'boolflag', '{default=yes}'))
+
+    for arg in args:
+        print('arg {number=%d}{call=%s}{display=%s}{tooltip=%s}{type=%s}%s' % arg)
+
+    values = values + [(Config.CHANNEL.value, '%d' % i, '%d' % i, 'true' if i == 11 else 'false') for i in range(11, 27)]
+
+    for value in values:
+        print('value {arg=%d}{value=%s}{display=%s}{default=%s}' % value)
+
+def extcap_dlts(interface):
+    """List DLTs for the given interface"""
+    print('dlt {number=195}{name=IEEE802_15_4_WITHFCS}{display=IEEE 802.15.4 with FCS}')
+    print('dlt {number=283}{name=IEEE802_15_4_TAP}{display=IEEE 802.15.4 TAP}')
+
+def serialopen(interface,  __console__, DirtylogFile):
+    """
+    Open serial to indentify OpenThread sniffer
+    :param interface: string, eg: '/dev/ttyUSB0 - Zolertia Firefly platform', '/dev/ttyACM1 - nRF52840 OpenThread Device'
+    """
+    sys.stdout = DirtylogFile
+    sys.stderr = DirtylogFile
+    interface = str(interface).split()[0]
+    baudrate = None
+    result = None
+
+    for speed in COMMON_BAUDRATE:
+        stream = StreamOpen('u', interface, False, baudrate=speed)
+
+        wpan_api = WpanApi(stream, nodeid=DEFAULT_NODEID)
+        # result is None for NCP, not None for RCP
+        result = wpan_api.prop_set_value(SPINEL.PROP_PHY_ENABLED, 1) # detect baudrate
+
+        # result should not be None for both NCP and RCP
+        result = wpan_api.prop_get_value(SPINEL.PROP_CAPS) # confirm OpenThread Sniffer
+
+        # check whether or not is OpenThread Sniffer
+        stream.close()
+        if result is None:
+            continue
+        else:
+            baudrate = speed
+            break
+
+    sys.stdout = __console__
+    sys.stderr = __console__
+
+    if baudrate is not None:
+        if sys.platform == 'win32':
+            # Wireshark only shows the value of key `display`('OpenThread Sniffer').
+            # Here intentionally appends interface in the end (e.g. 'OpenThread Sniffer: COM0').
+            print('interface {value=%s:%s}{display=OpenThread Sniffer %s}' % (interface, baudrate, interface))
+        else:
+            # On Linux or MacOS, wireshark will show the concatenation of the content of `display`
+            # and `interface` by default (e.g. 'OpenThread Sniffer: /dev/ttyACM0').
+            print('interface {value=%s:%s}{display=OpenThread Sniffer}' % (interface, baudrate))
+
+def extcap_interfaces():
+    """List available interfaces to capture from"""
+    __console__ = sys.stdout
+
+    DirtylogFile = open(os.path.join(tempfile.gettempdir(), 'dirtylog'), 'w')
+    print('extcap {version=1.0.0}{display=OpenThread Sniffer}{help=https://github.com/openthread/pyspinel}')
+
+    for interface in comports():
+        th = threading.Thread(target=serialopen, args=(interface, __console__, DirtylogFile))
```

Comment:
Multiple threads are running `serialopen`, and all `serialopen` will operate sys.stdout, sys.stderr. 
Won't there be data race?

Reply:
My previous thought was to redirect this part of the dirtylog to a file, but now it seems that this is not a good solution.
I want to cancel this part of the content, and instead modify the content of the dirtylog in the spinel, such as not determining whether an array subscript is out of bounds.
What do you think of this idea?

Reply:
Agree that there is room for further enhancements when comes to multi-thread.
Also agree that we may need to manage the output of traceback in spinel implementation, and I would prefer  keeping original behavior by default.
How about defining a variable e.g. `TRACE_LOGFILE`(default disabled), like existing `DEBUG_ENABLE` in config.py? 
This switch would be enabled when the sniffer is initiated by extcap,  the trace logs will be output to the some logfile specific to the sniffer instance.

@DangZRQ @simonlingoogle thoughts?